server/src/services/blogs/service.js

```diff
@@ -0,0 +1,83 @@
+import mongoose from 'mongoose';
+import Blogs from '../../models/blogs';
+
+const saveBlog = (req, res, next) => {
+  const { title } = req.body;
+  const { content } = req.body;
+  const userId = req.user._id;
+
+  const blog = {
+    title,
+    content,
+    userId,
+  };
+
+  Blogs.create(blog, (err, blogFromDB) => {
+    if (err) {
+      res.status(500).json({ msg: 'Something went wrong' });
+    }
+    req.blog = blogFromDB;
+    next();
+  });
+};
+
+const fetchBlog = (req, res, next) => {
+  const blogID = req.params.id;
+  const validObjectId = mongoose.Types.ObjectId.isValid(blogID);
+
+  if (validObjectId) {
+    Blogs.findById(blogID, (err, blog) => {
+      if (err) {
+        return res.status(500).json({ msg: 'Something went wrong' });
+      }
+      if (blog) {
+        req.blog = blog.toObject();
+        next();
+      } else {
+        return res.status(404).json({ msg: "can't find blog with the given id." });
```

Comment:
The message can be replaced by Blog not found. Just a thought though this error will work as well. :smile: